libraries/aws/provisioning/test/system/aws_iot_tests_provisioning_system.c

```diff
@@ -493,3 +648,73 @@ TEST( Provisioning_System, RegisterThingNominalCase )
     IotTest_Free( newCertificateContext.pCertificateIdBuffer );
```

Comment:
What happens if the malloc failed in _storeDataFromKeysAndCertResponseCallback and we free a NULL buffer is this ok?

Reply:
I think that the free operation would accept NULL pointers. But better to be safe than sorry with cryptic TravisCI test failures, so have added if-checks around the buffers before calling `Free`